catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/AbstractAssociatedMetacardsHandler.java

```diff
@@ -0,0 +1,149 @@
+
+  protected CatalogFramework catalogFramework;
+
+  protected EndpointUtil util;
+
+  protected FilterBuilder filterBuilder;
+
+  public AbstractAssociatedMetacardsHandler(
+      CatalogFramework catalogFramework, EndpointUtil util, FilterBuilder filterBuilder) {
+    this.catalogFramework = catalogFramework;
+    this.util = util;
+    this.filterBuilder = filterBuilder;
+  }
+
+  public void handle(Metacard existing, Metacard updated, List<Metacard> queries) {
+    create(existing, updated, queries);
+    delete(existing, updated, queries);
+    update(existing, updated, queries);
+  }
+
+  abstract void create(Metacard existing, Metacard updated, List<Metacard> metacards);
+
+  abstract void update(Metacard existing, Metacard updated, List<Metacard> metacards);
+
+  abstract void delete(Metacard existing, Metacard updated, List<Metacard> metacards);
+
+  List<Serializable> getMetacardIds(List<Metacard> metacards) {
+    return metacards.stream().map(Metacard::getId).collect(Collectors.toList());
+  }
+
+  List<Serializable> getAssociatedMetacardIds(Metacard workspace, String association) {
+    Attribute attr = workspace.getAttribute(association);
+    return attr == null ? new ArrayList<>() : attr.getValues();
+  }
+
+  boolean isAssociated(Metacard workspace, Metacard metacard, String association) {
+    return getAssociatedMetacardIds(workspace, association).contains(metacard.getId());
+  }
+
+  Metacard getMetacard(String id) {
+    try {
+      return util.getMetacard(id);
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      LOGGER.warn("Failed to retrieve query metacard {}", id, e);
+    }
+
+    return null;
+  }
+
+  Metacard createMetacard(Metacard metacard) {
+    LOGGER.trace("Creating new query metacard {}", metacard.getId());
+
+    try {
+      return catalogFramework.create(new CreateRequestImpl(metacard)).getCreatedMetacards().get(0);
+    } catch (IngestException | SourceUnavailableException e) {
+      LOGGER.debug("Query metacard could not be ingested {}", e);
+    }
+
+    return null;
+  }
+
+  Metacard updateMetacard(String id, Metacard metacard) {
+    try {
+      return catalogFramework
+          .update(new UpdateRequestImpl(id, metacard))
+          .getUpdatedMetacards()
+          .get(0)
+          .getNewMetacard();
+    } catch (IngestException | SourceUnavailableException e) {
+      LOGGER.debug("Query metacard could not be updated {}", id, e);
+    }
+
+    return null;
+  }
+
+  Metacard deleteMetacard(String id) {
+    LOGGER.trace("Deleting query metacard {}", id);
+
+    try {
+      return catalogFramework.delete((new DeleteRequestImpl(id))).getDeletedMetacards().get(0);
+    } catch (IngestException | SourceUnavailableException e) {
+      LOGGER.debug("Query metacard could not be deleted {}", id, e);
+    }
+
+    return null;
+  }
+
+  boolean hasChanges(String id, Metacard updated, Set<String> attributes) {
```

Comment:
Could potentially reuse the path endpoint for workspace

Reply:
@bellcc Here's the thing i mentioned
https://github.com/codice/ddf/blob/master/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java#L968-L1009

Reply:
Yeah `patchMetacards` does something similar however I prefer this method because I want to be able to detect specific changes to attributes